src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1769,6 +1769,21 @@ public ErrorDiagnostic NestedRuntimePropertyAccessNotSupported(string? resourceS
                 TextSpan,
                 "BCP311",
                 $@"The provided index value of ""{indexSought}"" is not valid for type ""{typeName}"". Indexes for this type must be between 0 and {tupleLength - 1}");
+
+            public ErrorDiagnostic CyclicArmTypeRefs(IEnumerable<string> cycleLinks) => new(
+                TextSpan,
+                "BCP312",
+                $@"Type reference cycle detected in the linked ARM JSON file (""{string.Join(" -> ", cycleLinks.Select(l => $@"""{l}"""))}"").");
```

Comment:
Could you explain how a type reference cycle occurs? Maybe an example would suffice

Reply:
A cycle happens when two or more symbols are defined in terms of each other:

```bicep
type fizz = buzz
type buzz = fizz
```

ARM template type references use local JSON pointer URIs, so you can end up with the same kind of circular definition:

```json
{
  ...
  "definitions": {
    "fizz": {
      "$ref": "#/definitions/buzz"
    },
    "buzz": {
      "$ref": "#/definitions/fizz"
    }
  }
}
```

Reply:
Makes sense thank you!